org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/DefaultBindingResolver.java

```diff
@@ -252,15 +249,14 @@ Util.BindingsToNodesMap getBindingsToNodesMap() {
 		return new Util.BindingsToNodesMap() {
 			@Override
 			public org.eclipse.jdt.internal.compiler.ast.ASTNode get(Binding binding) {
-				return (org.eclipse.jdt.internal.compiler.ast.ASTNode)
-					DefaultBindingResolver.this.newAstToOldAst.get(DefaultBindingResolver.this.bindingsToAstNodes.get(binding));
+				return DefaultBindingResolver.this.newAstToOldAst.get(DefaultBindingResolver.this.bindingsToAstNodes.get(binding));
```

Comment:
There is a warning here: "Unlikely argument type Binding for get(Object) on a Map<IBinding,ASTNode>" . I see you're not changing the code here so is this uncovering a bug that has been lurking?

Reply:
I think it's uncovering a bug